GitUI/Shells/ShellDescriptor.cs

```diff
@@ -0,0 +1,33 @@
+﻿#nullable enable
+
+using System.Drawing;
+
+namespace GitUI.Shells
+{
+    public abstract class ShellDescriptor : IShellDescriptor
+    {
+        /// <inheritdoc/>
+        public string? ExecutableCommandLine { get; protected set; } = null!;
+
+        /// <inheritdoc/>
+        public string ExecutableName { get; protected set; } = null!;
+
+        /// <inheritdoc/>
+        public string? ExecutablePath { get; protected set; } = null!;
+
+        /// <inheritdoc/>
+        public bool HasExecutable => ExecutablePath is object;
```

Comment:
```suggestion
        public bool HasExecutable => ExecutablePath != null;
```

Reply:
It would be good to write `not is null`, but this is C# 9.0 feature, which hasn't been released yet.
So `is object` or `is {}` is a replacement for `!= null`.

Reply:
So let's wait for `not is null`. `is {}` is write-only, `is object` is not much clearer and smells like dynamic cast, although the compiler will be able to optimize it.

Reply:
No worries at all. 
A correction, once VS 16.7 (not C# 9) is shipped we can use `not is null`